src/ui/states/state-needs-payment-info.svelte

```diff
@@ -197,21 +111,52 @@
   };
 
   const localeToUse = getLocaleToUse(stripeElementLocale);
-</script>
 
-<div class="checkout-container">
-  <!-- <TextSeparator text="Pay by card" /> -->
-  {#if stripe && clientSecret}
-    <form on:submit|preventDefault={handleContinue}>
-      <Elements
-        {stripe}
-        {clientSecret}
-        loader="always"
-        locale={localeToUse}
-        bind:elements
-        theme="stripe"
-        labels="floating"
-        variables={{
+  onMount(() => {
+    updateStripeVariables();
+
+    window.addEventListener("resize", onResize);
+
+    return () => {
+      window.removeEventListener("resize", onResize);
+    };
+  });
+
+  $: {
+    // @ts-ignore
+    if (elements && elements._elements.length > 0) {
+      safeElements = elements;
+    }
+  }
+
+  onMount(async () => {
+    const stripePk = paymentInfoCollectionMetadata.data.publishable_api_key;
+    const stripeAcctId = paymentInfoCollectionMetadata.data.stripe_account_id;
+
+    if (!stripePk || !stripeAcctId) {
+      throw new Error("Stripe publishable key or account ID not found");
+    }
+
+    stripe = await loadStripe(stripePk, {
+      stripeAccount: stripeAcctId,
+    });
+
+    if (!stripe) {
+      throw new Error("Stripe client not found");
+    }
+
+    elements = stripe.elements({
+      loader: "always",
+      locale: localeToUse,
+      mode: "payment",
+      paymentMethodTypes: ["card"],
+      setup_future_usage: "off_session",
+      amount: productDetails.currentPrice.amountMicros / 10000,
+      currency: productDetails.currentPrice.currency.toLowerCase(),
```

Comment:
Since these parameters should match the settings from the PaymentIntent/SetupIntent, I was considering including them in the `CheckoutStart` response. Wdyt @RevenueCat/web?

Reply:
I changed this to retrieve these values from the start purchase response so that we have more control over them from the backend